src/octoclass.cpp

```diff
@@ -520,7 +542,7 @@ void OctoClass::FadeMemory(const double &rate) {  // rate at which this function
 
         // tree nodes that are unknown
         if (is_occ != tree_inflated_.isNodeOccupied(n)) {  // if it was occupied then disoccupied, delete node
-            tree_inflated_.deleteNode(key, it.getDepth());
+            tree_inflated_.deleteNode(key, depth);
```

Comment:
I think is better not to change `it.getDepth()`  in this line, since you define `depth` as a const unint. on the other hand you could define, depth as a pointer to `it.getDepth()` , instead of making a copy of this iterator.

Reply:
Sorry, I don't follow your comment... what is the problem of having `depth` as a `const uint`?
If you are worried about complexity, `uint` should take less RAM than a pointer, as `uint` has 4 bytes, while a pointer has 8 bytes (in a 64bit machine).

Reply:
it's a trivial matter, you declare `depth` variable in line 527 and it's not changing anymore.  but technically `it.getDepth()` iterator can check the most recent root nodes.

Reply:
not really... the iterator can only point to a different node after if moves on in the `for` loop through the command `++it`...

Reply:
is this thread-safe? If not then potentially it could change?

Reply:
This is the only place where `FadeMemory` is called:
```
        mutexes_.octomap.lock();
            if (globals_.octomap.memory_time_ > 0) {
                globals_.octomap.FadeMemory(fading_memory_update_rate_);
            }
        mutexes_.octomap.unlock();
```
So yeah, it is thread-safe